sunpy/map/tests/test_header_helper.py

```diff
@@ -224,19 +225,31 @@ def test_make_fitswcs_header_handles_dn(input_unit, output_string, map_data, hpc
     header = make_fitswcs_header(map_data, hpc_coord, unit=input_unit)
     assert header['bunit'] == output_string
 
-
-def test_invalid_inputs(map_data, hcc_coord, hpc_coord_notime, hpc_coord):
-    # Raise the HCC error
-    with pytest.raises(ValueError):
-        make_fitswcs_header(map_data, hcc_coord)
-
-    # Check for when coordinate argument isn't given as an `astropy.coordinate.SkyCoord`
-    with pytest.raises(ValueError):
-        make_fitswcs_header(map_data, map_data)
-
-    # Check for when an observation time isn't given
-    with pytest.raises(ValueError):
-        make_fitswcs_header(map_data, hpc_coord_notime)
+error_messages_1 = {
+    "heliocentric_unsupported": "This function does not currently support heliocentric coordinates.",
+    "coord_instance": "coordinate needs to be a coordinate frame or an SkyCoord instance.",
+    "missing_obstime": "The coordinate needs an observation time, `obstime`."
+}
+
+@pytest.mark.parametrize(
+    ("coordinate_input", "error_messages_1"),
+    [
+        ("hcc_coord", error_messages_1["heliocentric_unsupported"]),   # Unsupported HCC error
+        ("map_data", error_messages_1["coord_instance"]),              # Invalid coordinate type
+        ("hpc_coord_notime", error_messages_1["missing_obstime"])      # Missing observation time
```

Comment:
Same here, why the dict and not just the error message?

Reply:
I resolved it in the previous query. However, in this case, I think it is best to leave it here as it is as it improves readability in my opinion.

Reply:
I honestly do not agree. You have a dict, which you have a keyword, then you also add the comment on what the error is. 

This is just doing double duty, this should be changed.